tests/neuro_san/client/agent_evaluator.py

```diff
@@ -0,0 +1,29 @@
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+
+from typing import Any
+
+from neuro_san.message_processing.basic_message_processor import BasicMessageProcessor
+
+
+class AgentEvaluator:
+    """
+    Interface definition for evaluating part of an agent's response
+    """
+
+    def evaluate(self, processor: BasicMessageProcessor, verify_for: Any):
```

Comment:
Interface for code that describes how we are going to test an aspect of agent response.

Worth noting that a BasicMessageProcessor is what is needed for now.
This might evolve in the future.